servicetalk-grpc-api/src/main/java/io/servicetalk/grpc/api/GrpcClientBuilder.java

```diff
@@ -281,8 +281,27 @@
      *
      * @return {@code this}
      * @see #unresolvedAddressToHost(Function)
+     * @deprecated Use {@link #hostHeaderFallback(boolean)}.
      */
-    public abstract GrpcClientBuilder<U, R> disableHostHeaderFallback();
+    @Deprecated
+    public GrpcClientBuilder<U, R> disableHostHeaderFallback() {
+        return hostHeaderFallback(false);
+    }
+
+    /**
+     * Configures automatically setting {@code Host} headers by inferring from the address or {@link HttpMetaData}.
+     * <p>
+     * When {@code false} is passed, this setting disables the default filter such that no {@code Host} header will be
+     * manipulated.
+     *
+     * @param enable Whether a default filter for inferring the {@code Host} headers should be added.
+     * @return {@code this}
+     * @see #unresolvedAddressToHost(Function)
+     */
+    public GrpcClientBuilder<U, R> hostHeaderFallback(boolean enable) {
+        throw new UnsupportedOperationException("Setting automatic host header fallback using this method" +
+                " is not yet supported.");
```

Comment:
In the other PR @bondolo used `getClass().getSimpleName()`: https://github.com/apple/servicetalk/pull/1781/files#diff-fa20d1d978b59de34bb9649cc6cc21ba5321886ccaa29636bbc285eea14eb07aR332
Consider using `getClass().getName()` moving forward. It will help users understand who they should contact if this method is not implemented.